core/sail/lucene-api/src/main/java/org/eclipse/rdf4j/sail/lucene/AbstractSearchIndex.java

```diff
@@ -855,6 +859,59 @@ private Iterable<? extends DocumentResult> evaluateQuery(GeoRelationQuerySpec qu
 		return hits;
 	}
 
+	private IRI toSpatialOp(String relation) {
+		if (GEOF.SF_INTERSECTS.stringValue().equals(relation)) {
+			return GEOF.SF_INTERSECTS;
+		} else if (GEOF.SF_DISJOINT.stringValue().equals(relation)) {
+			return GEOF.SF_DISJOINT;
+		} else if (GEOF.SF_EQUALS.stringValue().equals(relation)) {
+			return GEOF.SF_EQUALS;
+		} else if (GEOF.SF_OVERLAPS.stringValue().equals(relation)) {
+			return GEOF.SF_OVERLAPS;
+		} else if (GEOF.EH_COVERED_BY.stringValue().equals(relation)) {
+			return GEOF.SF_WITHIN;
+		} else if (GEOF.EH_COVERS.stringValue().equals(relation)) {
+			return GEOF.EH_CONTAINS;
```

Comment:
Seems that everything else maps to itself expect for these two. Could you add a comment explaining why?

Reply:
The problem with this part is similar to this check:

https://github.com/ate47/rdf4j/blob/GH-5277-bad-geo-lucene-answers/core/sail/lucene/src/main/java/org/eclipse/rdf4j/sail/lucene/impl/LuceneIndex.java#L915

Which is badly made and would require another issue/pr.